### PR TITLE
Switch to `package:dart_flutter_team_lints`

### PR DESCRIPTION
**Pull request by mosuem**
---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by jakemac53 — COMMENTED** [review]
Even though there aren't meaningful changes, we should update the version to a `-wip` one indicate there have been changes. The changelog entry can just be empty

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,5 +1,14 @@
-include: package:lints/recommended.yaml
+include: package:dart_flutter_team_lints/analysis_options.yaml
 
+analyzer:
+  errors:
+    avoid_dynamic_calls: ignore
```
Comment (jakemac53) [inline analysis_options.yaml:5]: We should probably file an issue to clean these up

**Comment by mosuem** [comment]
`dart pub get` is failing because of `package:test`, which has a pubspec entry:
```yaml
 # Use a tight version constraint to ensure that a constraint on matcher
 # properly constrains all features it provides.
 matcher: '>=0.12.16 <0.12.17'
```

Does this mean we need to upgrade the dependency in `test` first? And why have this tight upper bound anyhow? Don't we have semver to handle this?

**Comment by jakemac53** [comment]
> Does this mean we need to upgrade the dependency in test first? 

We should instead just add a dependency_override in this package, on the `test` package. It might need to list a few of the related packages as well (test_api, test_core).

> Does this mean we need to upgrade the dependency in `test` first? And why have this tight upper bound anyhow? Don't we have semver to handle this?

This is because package:test exports package:matcher.

**Comment by mosuem** [comment]
Closing as the [dart-lang/matcher](https://github.com/dart-lang/matcher) repository is merged into the [dart-lang/test](https://github.com/dart-lang/test) monorepo. Please re-open this PR there!

**Pull request closed without merging**